### ISSUE TITLE
upside down background in my result Images

### ISSUE THREAD

**Issue by FarzanehCS** [karansher/computer-graphics-ray-casting#20]
Hi,
I am getting the following image, instead of the ones shown on the A2's repo!

I get this:

![image](https://user-images.githubusercontent.com/34817738/192112189-1dead191-6687-46b1-9a15-2f227bd8adcd.png)

VS this correct one in the repo:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/34817738/192112212-72ad6c2e-ba6f-475d-8938-543381d0e3ae.png">

Not sure why! I am guessing the problem could be how I create the ray vector. I use this vector as the d in this formula and then multiply it by the transformation matrix. Similar to:

<img width="741" alt="image" src="https://user-images.githubusercontent.com/34817738/192112282-96d29697-db5b-4be2-9bcc-374a21cab122.png">

So I am using the vector [u_i, v_j, -d] is used as it says (i.e. I use `u, v, -camera.d`). Also, I am aware that in our case we should use u_j, and u_i instead.

What could be the problem here?

**Comment by Ming-Yeung-Alfred-Meng** [comment]
The top half of the pixels detect the blue plane which is actually below the sphere...that means there is something wrong with the t value of your ray at detection. There are two possibilities/cases, 1) your rays point toward the correct direction, -w, then in plane's intersect(), you are only allowing negative values of t, 2) your rays point toward the opposite direction, then fix in it viewing_ray.

**Comment by FarzanehCS** [comment]
So I see if I use `u, -v, -camera.d`, i.e. if I use `-v`, the issue is solved. But I am not sure why? Is that because we consider the top left as the origin of the image plane? I still do not understand why?

**Comment by panuelosj** [comment]
Yes, it's because we take the (0,0) pixel to be the top left with positive values pointing down, while in the camera space basis, positive v points up.
